pages/studentPortal/web/React/assignments/template.tsx

```diff
@@ -1,15 +1,48 @@
+import { useQuery } from "@apollo/client";
 import { Router, useRouter } from "next/router";
 
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 
 import AssignmentComponent, {
   AssignmentComponentData,
   Stage,
 } from "../../../../../components/coding/studentPortal/AssignmentComponent";
 import { Button } from "../../../../../components/ui/Button";
+import {
+  FETCH_USER_ASSIGNMENT_SUBMISSIONS,
+  FetchUserAssignmentSubmissionsDataResponse,
+  UserAssignmentSubmissionsData,
+} from "../../../../../graphql/fetchUserAssignmentSubmissions";
+import { useAuth } from "../../../../../lib/authContext";
 
 const React2 = ({ incompleteStage, submittedStage, completedStage }) => {
   const router = useRouter();
+
+  const { user } = useAuth();
+  const [assignments, setAssignments] = useState<
+    UserAssignmentSubmissionsData[]
+  >([]);
+
+  const { data } = useQuery<FetchUserAssignmentSubmissionsDataResponse>(
+    FETCH_USER_ASSIGNMENT_SUBMISSIONS,
+    {
+      variables: {
+        user_id: user.uid,
+      },
+
+      onCompleted: (data: FetchUserAssignmentSubmissionsDataResponse) => {
+        setAssignments(data.user_assignment_submissions);
+      },
+    }
+  );
+  useEffect(() => {
+    if (assignments) {
+      setStage(Stage.SUBMITTED);
+    } else {
```

Comment:
Is it better to use the useEffect here or just to put in the onCompleted as well?